.travis.yml

```diff
@@ -1,76 +1,83 @@
+language: rust
+sudo: required
+dist: trusty
+services:
+  - docker
+
 matrix:
   include:
+    # Linux builds use the `rust-slave-dist` image so we link them against a
+    # "super old glibc" to ensure that it runs on as many platforms as possible.
+    - os: linux
+      rust: nightly
+      env: TARGET=i686-unknown-linux-gnu
+           DOCKER=alexcrichton/rust-slave-dist:2015-10-20b
+    - os: linux
+      rust: nightly
+      env: TARGET=x86_64-unknown-linux-gnu
+           DOCKER=alexcrichton/rust-slave-dist:2015-10-20b
+
+    # Cross builds happen in the `rust-slave-linux-cross` image to ensure that
+    # we use the right cross compilers for these targets. That image should
+    # bundle all the gcc cross compilers to enable us to build OpenSSL
+    - os: linux
+      rust: nightly
+      env: TARGET=arm-unknown-linux-gnueabi
+           DOCKER=alexcrichton/rust-slave-linux-cross:2016-03-17b
+           SKIP_TESTS=1
```

Comment:
I think? I've had lots of trouble in the past with QEMU's user-space emulation (e.g. not run a full emulator) with programs that use threads. In theory though we could just run the whole binary in an emulator (similar to how the libc crate is working). Think we should try that before landing here? It'd certainly make rustup the most heavily tested project after libc :P